src/efftester.ml

```diff
@@ -104,64 +114,41 @@ type term =
   | Let of variable * etype * term * term * etype * eff
   | If of etype * term * term * term * eff
 
-
 (** Printing functions  *)
-      
-let rec typeToOCaml ?(effannot=false) sb = function
-  | Typevar a  -> Buffer.add_string sb ("'a" ^ (string_of_int a))
-  | Unit       -> Buffer.add_string sb "unit"
-  | Int        -> Buffer.add_string sb "int"
-  | Bool       -> Buffer.add_string sb "bool"
-  | String     -> Buffer.add_string sb "string"
-  | List s     ->
-    (Buffer.add_string sb "(";
-     typeToOCaml ~effannot:effannot sb s;
-     Buffer.add_string sb ") list")
-  | Fun (s, e, t) ->
+
+let rec typeToOCaml ?(effannot = false) sb = function
+  | Typevar a -> Printf.bprintf sb "'a%d" a
+  | Unit -> Buffer.add_string sb "unit"
+  | Int -> Buffer.add_string sb "int"
+  | Bool -> Buffer.add_string sb "bool"
+  | String -> Buffer.add_string sb "string"
```

Comment:
Nitpick: I would recommend to not mix `Printf` and `Buffer` functions for consistency, using `Printf.bprintf sb "string"` in that case.